### ISSUE TITLE
Storybook is not applying styles when working on components

### ISSUE THREAD

**Issue by jacebenson** [redwoodjs/redwood#4361]
When I use `yarn rw storybook` the styles applied from my project are not applied in storybook to the component.  I'm not sure how to set that up or even where that is?

**Comment by thedavidprice** [comment]
@jacebenson I can't reproduce an error locally on my test project. I'm running `yarn rw setup ui tailwindcss` followed by `yarn rw sb` and everything works fine.

Can you share your tailwindcss config?

Also, can you paste the output from `yarn rw info`?

**Comment by jacebenson** [comment]
I'll see if I can recreate it with a fresh project. This came up in a discussion on maker hour.

**Comment by dthyresson** [comment]
@thedavidprice  I can see this with v0.43

![image](https://user-images.githubusercontent.com/1051633/152578923-abb65f82-c477-4224-9a32-27bd6c3b2eb5.png)

vs

![image](https://user-images.githubusercontent.com/1051633/152579009-ce0bb7ef-bd5c-4d11-ab68-0f9baeae7dbe.png)


`yarn rw info`

```
  System:
    OS: macOS 12.1
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - /var/folders/ft/8320mthj6gbdd2pmc42x13480000gn/T/yarn--1643997422423-0.6035660934236191/node
    Yarn: 1.22.17 - /var/folders/ft/8320mthj6gbdd2pmc42x13480000gn/T/yarn--1643997422423-0.6035660934236191/yarn
  Databases:
    SQLite: 3.36.0 - /usr/bin/sqlite3
  Browsers:
    Chrome: 98.0.4758.80
    Safari: 15.2
  npmPackages:
    @redwoodjs/core: v0.43.0 => 0.43.0 

✨  Done in 2.44s.
```

`tailwind config`:

```
module.exports = {
  content: ['src/**/*.{js,jsx,ts,tsx}'],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

**Comment by simoncrypta** [comment]
I already saw this problem before, I think every work when it uses with JIT. I think @jtoar can confirm this ?

**Comment by dthyresson** [comment]
I tried the following tailwind config with and without purge options:


```js
module.exports = {
  content: ['src/**/*.{js,jsx,ts,tsx}'],
  mode: 'jit',
  // purge: ['./src/**/*.{js,jsx,ts,tsx}'],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

But still no style rendering in storybook.

Could it be the way RW scaffold use tailwind scaffolds styles that use `@apply`?

**Comment by dthyresson** [comment]
> Could it be the way RW scaffold use tailwind scaffolds styles that use `@apply`?

Yup, that's the issue. If you use tailwind css in your component like:

```
      <div className="bg-slate-500 p-12">
...
```

then

<img width="606" alt="image" src="https://user-images.githubusercontent.com/1051633/152656003-9d545bcb-29be-45e2-ac0a-2ec378072bb7.png">


The styling applied.

So, the issue is that scaffold that use the RW tailwind stylesheet styles are not rending in storybook.

**Comment by thedavidprice** [comment]
@dthyresson Good work diagnosing this one! Do you have suggestions about next steps. Some reactions on my part:
- I'm not sure how to prioritize this; Storybook for Scaffolds doesn't seem important to me
- ...unless someone is modifying and gets confused 'cause they won't work
- reworking the tailwind Scaffold css template to _not_ use `@apply` makes the most sense if we try to "fix" this; seems like a lot of work
- most straightforward for me is just visibility about what's happening so people can resolve if/as needed

Thoughts?

**Comment by simoncrypta** [comment]
> * Storybook for Scaffolds doesn't seem important to me

I agree with you, I think it should only be communicated

**Comment by virtuoushub** [comment]
> When I use `yarn rw storybook` the styles applied from my project are not applied in storybook to the component. I'm not sure how to set that up or even where that is?


@jacebenson What about [making/editing a `web/config/storybook.preview.js`](https://redwoodjs.com/docs/storybook#configuring-rendering-with-storybookpreviewjs) to include

```js
import '../src/scaffold.css'
import '../src/index.css'
```

?

**Comment by william-stafflink** [comment]
Hey all I am having the same issue and have tried the solution above to no avail... Has anyone been able to find a workaround   ?

**Comment by william-stafflink** [comment]
> Hi @william-stafflink do you have a public repo I can take a look at where this is happening?
> 
> regardless, a simple check is to confirm the `web/src/index.css` file exists. and contains something like these lines:
> 
> ```css
> ...
> /**
>  * START --- SETUP TAILWINDCSS EDIT
>  *
>  * `yarn rw setup ui tailwindcss` placed these imports here
>  * to inject Tailwind's styles into your CSS.
>  * For more information, see: https://tailwindcss.com/docs/installation#include-tailwind-in-your-css
>  */
> @import 'tailwindcss/base';
> @import 'tailwindcss/components';
> @import 'tailwindcss/utilities';
> /**
>  * END --- SETUP TAILWINDCSS EDIT
>  */
> ...
> ```

Hi @virtuoushub unfortunately no public repo. Running Index.css exactly as the above
```css
/**
 * START --- SETUP TAILWINDCSS EDIT
 *
 * `yarn rw setup ui tailwindcss` placed these imports here
 * to inject Tailwind's styles into your CSS.
 * For more information, see: https://tailwindcss.com/docs/installation#include-tailwind-in-your-css
 */
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
/**
 * END --- SETUP TAILWINDCSS EDIT
 */
.player-wrapper {
  position: relative;
  padding-top: 56.25%; /* 720 / 1280 = 0.5625 */
}

.react-player {
  position: absolute;
  top: 0;
  left: 0;
}
```
and Importing it from `web/src/styles/index.css` and have tried running from `web/src/index.css`

**Comment by carlosblanc0** [comment]
[TSX] Ran into the exact same issue this morning. my workaround is that instead of using this:

> I tried the following tailwind config with and without purge options:
> 
> ```js
> module.exports = {
>   content: ['src/**/*.{js,jsx,ts,tsx}'],
>   mode: 'jit',
>   // purge: ['./src/**/*.{js,jsx,ts,tsx}'],
>   theme: {
>     extend: {},
>   },
>   plugins: [],
> }
> ```
> 
> But still no style rendering in storybook.
> 
> Could it be the way RW scaffold use tailwind scaffolds styles that use `@apply`?

I went with this:
> 
> ```js
> module.exports = {
>   content: './src/**/*.tsx',
>  // mode: 'jit',
>   // purge: ['./src/**/*.{js,jsx,ts,tsx}'],
>   theme: {
>     extend: {},
>   },
>   plugins: [],
> }
> ```
> 

Cleared cache, restarted both servers (redundant, I know), and it's now working as expected.

>Software:

    System Software Overview:

      System Version: macOS 12.6 (21G115)
      Kernel Version: Darwin 21.6.0
      Boot Volume: Macintosh HD - Data
      Boot Mode: Normal
      Computer Name: Carlos’s MacBook Pro
      User Name: Carlos A. Blanco (carlosa.blanco)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 3 days 2:18

>Hardware:

    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro13,3
      Processor Name: Quad-Core Intel Core i7
      Processor Speed: 2.6 GHz
      Number of Processors: 1
      Total Number of Cores: 4
      L2 Cache (per Core): 256 KB
      L3 Cache: 6 MB
      Hyper-Threading Technology: Enabled
      Memory: 16 GB
      System Firmware Version: 451.140.1.0.0
      OS Loader Version: 540.120.3~22
      SMC Version (system): 2.38f12
>

> Environment

> 
   "devDependencies": {
    "@babel/core": "^7.19.3",
    "@storybook/addon-actions": "^6.5.12",
    "@storybook/addon-essentials": "^6.5.12",
    "@storybook/addon-interactions": "^6.5.12",
    "@storybook/addon-links": "^6.5.12",
    "@storybook/builder-vite": "^0.2.4",
    "@storybook/react": "^6.5.12",
    "@storybook/testing-library": "^0.0.13",
    "@types/react": "^18.0.17",
    "@types/react-dom": "^18.0.6",
    "@vitejs/plugin-react": "^2.1.0",
    "autoprefixer": "^10.4.12",
    "babel-loader": "^8.2.5",
    "postcss": "^8.4.18",
    "tailwindcss": "^3.1.8",
    "typescript": "^4.6.4",
    "vite": "^3.1.0"
  }
>

**Comment by kaustavkarmakar2** [comment]
Storybook is not taking inline tailwind css...like <div className="bg-purple"></div>

**Comment by SegopotsoMakhutja** [comment]
if anyone runs into this issue again and they've configured a `prefix` for their classNames in tailwind.config, consider removing it. for me that was the reason the styles were not being applied.
